src/imp/security_framework.rs

```diff
@@ -82,6 +84,41 @@ pub struct Identity {
 }
 
 impl Identity {
+    pub fn from_pkcs8(pem: &[u8], key: &[u8]) -> Result<Identity, Error> {
+        if !key.starts_with(b"-----BEGIN PRIVATE KEY-----") {
+            return Err(Error(base::Error::from(errSecParam)));
+        }
+
+        let dir = TempDir::new().map_err(|_| Error(base::Error::from(errSecIO)))?;
+        let keychain = keychain::CreateOptions::new()
+            .password("password")
```

Comment:
It would probably be better to generate a random password for each of these keychains since they will be visible on disk for a small period of time.

Reply:
Done. By the way, would you accept a separate PR after this to use the 2018 edition?

Reply:
Sure